test/krb5/mzcompose.yml

```diff
@@ -10,115 +10,85 @@
 # Portions of this file are derived from
 # https://github.com/vdesabou/kafka-docker-playground/blob/master/environment/kerberos/docker-compose.yml
 
----
 version: "3.7"
 services:
   kdc:
     build: kdc
-    hostname: kdc.krb5.local
     volumes:
       - secrets:/var/lib/secret
       - ./kdc/krb5.conf:/etc/krb5.conf
 
   zookeeper:
     image: confluentinc/cp-zookeeper:latest
-    # Kafka seems to insist on authenticating to the zookeeper hostname that
-    # comes back from a reverse DNS lookup of its `KAFKA_ZOOKEEPER_CONNECT` host
-    # value. The only way to coordinate this with the desired hostname is to
-    # control the container name, as well.
-    container_name: zookeeper
-    hostname: zookeeper.krb5.local
     depends_on:
       - kdc
     volumes:
       - secrets:/var/lib/secret
       - ./kdc/krb5.conf:/etc/krb5.conf
-      - ./zookeeper.sasl.jaas.config:/etc/kafka/zookeeper.sasl.jaas.config
+      - ./sasl.jaas.config:/etc/zookeeper/sasl.jaas.config
     environment:
       ZOOKEEPER_CLIENT_PORT: 2181
-      ZOOKEEPER_TICK_TIME: 2000
-      KAFKA_OPTS: -Dzookeeper.4lw.commands.whitelist=*
-        -Djava.security.auth.login.config=/etc/kafka/zookeeper.sasl.jaas.config
-        -Djava.security.krb5.conf=/etc/krb5.conf
+      # Despite the environment variable name, these are JVM options that are
+      # passed through to Zookeeper.
+      KAFKA_OPTS: >-
         -Dzookeeper.authProvider.1=org.apache.zookeeper.server.auth.SASLAuthenticationProvider
+        -Dzookeeper.sessionRequireClientSASLAuth=true
+        -Djava.security.auth.login.config=/etc/zookeeper/sasl.jaas.config
+        -Djava.security.krb5.conf=/etc/krb5.conf
         -Dsun.security.krb5.debug=true
-        -Dzookeeper.allowSaslFailedClients=false
-        -Dzookeeper.requireClientAuthScheme=sasl
 
-  broker:
+  kafka:
     image: confluentinc/cp-kafka:latest
-    hostname: broker.krb5.local
     depends_on:
       - kdc
       - zookeeper
     volumes:
       - secrets:/var/lib/secret
       - ./kdc/krb5.conf:/etc/krb5.conf
-      - ./broker.sasl.jaas.config:/etc/kafka/broker.sasl.jaas.config
+      - ./sasl.jaas.config:/etc/kafka/sasl.jaas.config
     environment:
       KAFKA_BROKER_ID: 1
       KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
-      KAFKA_CONFLUENT_LICENSE_TOPIC_REPLICATION_FACTOR: 1
-      KAFKA_ADVERTISED_LISTENERS: SASL_PLAINTEXT://broker.krb5.local:9092
+      KAFKA_ADVERTISED_LISTENERS: SASL_PLAINTEXT://kafka:9092
       KAFKA_INTER_BROKER_LISTENER_NAME: SASL_PLAINTEXT
-      KAFKA_SASL_ENABLED_MECHANISMS: GSSAPI
       KAFKA_SASL_KERBEROS_SERVICE_NAME: kafka
-      KAFKA_ZOOKEEPER_CONNECT: zookeeper.krb5.local:2181
-      KAFKA_OPTS: -Djava.security.auth.login.config=/etc/kafka/broker.sasl.jaas.config
+      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
+      KAFKA_OPTS: >-
+        -Dzookeeper.sasl.client.canonicalize.hostname=false
```

Comment:
Gah--where did you find this setting?! Google was not offering up this as a suggestion for my many queries.

Reply:
It's buried deep, deep, deep in this absolutely horrible wiki page:

https://cwiki.apache.org/confluence/display/ZOOKEEPER/Client-Server+mutual+authentication